src/tests/java/picard/analysis/TheoreticalSensitivityTest.java

```diff
@@ -0,0 +1,238 @@
+        final double empiricalProportionWithinOneSigma = proportions.get(0).get(0) - proportions.get(0).get(1);
+
+        //the proportion within one sigma for the normal distribution
+        //hence whether any element falls within one sigma is a Bernoulli variable
+        final double theoreticalProportionWithinOneSigma = 0.682689492;
+        final double samplingStandardDeviationOfProportion = Math.sqrt(theoreticalProportionWithinOneSigma*(1-theoreticalProportionWithinOneSigma) /  sampleSize);
+
+        Assert.assertEquals(empiricalProportionWithinOneSigma, theoreticalProportionWithinOneSigma, 5*samplingStandardDeviationOfProportion);
+    }
+
+    //Put it all together for deterministic quality and depths
+    @Test
+    public void testDeterministicQualityAndDepth() throws Exception {
+        final double logOddsThreshold = 0.0;
+        final double tolerance = 0.001;
+        final int sampleSize = 1; //quality is deterministic, hence no sampling error
+        for (int q = 5; q < 10; q++) {
+            for (int n = 5; n < 10; n++) {
+                final double minAltCount = 10*n*Math.log10(2)/q;  //alts required to call when log odds ratio threshold = 1
+                double expectedResult = 0.0;
+
+                final List<ArrayList<Double>> altCountProbabilities = TheoreticalSensitivity.hetAltDepthDistribution(n+1);
+                for (int altCount = n; altCount > minAltCount; altCount--) {
+                    expectedResult += altCountProbabilities.get(n).get(altCount);
+                }
+
+                //deterministic weights that always yield q are 0.0 for 0 through q - 1 and 1.0 for q
+                final double[] qualityDistribution = new double[q+1];
+                Arrays.fill(qualityDistribution, 0L);
+                qualityDistribution[qualityDistribution.length-1]=1L;
+                final double[] depthDistribution = new double[n+1];
+                Arrays.fill(depthDistribution, 0L);
+                depthDistribution[depthDistribution.length-1]=1L;
+
+                final double result = TheoreticalSensitivity.hetSNPSensitivity(depthDistribution, qualityDistribution, sampleSize, logOddsThreshold);
+                Assert.assertEquals(result, expectedResult, tolerance);
+            }
+        }
+    }
+
+    @Test
+    public void testHetSensDistributions() throws Exception {
+        //Expect theoretical sens to be close to .9617 for Solexa-332667
+        final double tolerance = 0.02;
+        final double expectedResult = .9617;
+        final int maxDepth = 500;
+        final double [] depthDistribution = new double[maxDepth+1];
+        final double [] qualityDistribution = new double[50];
+
+        final Scanner scanDepth = new Scanner(DEPTH);
+        int i = 0;
+        for (i = 0; scanDepth.hasNextDouble(); i++) {
+            depthDistribution[i] = scanDepth.nextDouble();
+        }
+        while (scanDepth.hasNextDouble()) {
```

Comment:
@davidbenjamin @lbergelson I can't find the comments that were here before I pushed again.. it was about using a for loop instead of a while loop..